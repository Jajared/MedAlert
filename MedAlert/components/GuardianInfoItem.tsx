import { MaterialCommunityIcons } from "@expo/vector-icons";
import { StyleSheet, Text, View, SafeAreaView, Modal, TouchableOpacity } from "react-native";
import React, { useState } from "react";

export default function GuardianInfoItem({ props, navigation, removeGuardian }) {
  const guardianInfo = props.item;
  const [isDeleteModalVisible, setDeleteModalVisible] = useState(false);

  return (
    <TouchableOpacity onPress={() => navigation.navigate("Patient Medication", { guardianId: guardianInfo.UserId, guardianName: guardianInfo.Name })} testID="GuardianInfoItem">
      <SafeAreaView style={styles.itemContainer}>
        {guardianInfo.Gender == "Female" ? <MaterialCommunityIcons name="face-woman-profile" size={24} color="black" style={{ flex: 1 }} /> : <MaterialCommunityIcons name="face-man-profile" size={24} color="black" style={{ flex: 1 }} />}
        <View style={styles.infoContainer}>
          <Text style={styles.name}>{guardianInfo.Name}</Text>
          <Text style={styles.email}>{guardianInfo.EmailAddress}</Text>
        </View>
        <View style={styles.actionContainer}>
          <TouchableOpacity style={styles.removeButton} onPress={() => setDeleteModalVisible(true)} testID="RemoveButton">
            <Text style={styles.removeText}>Remove</Text>
          </TouchableOpacity>
        </View>
        <Modal visible={isDeleteModalVisible} transparent={true} animationType="slide" testID="DeleteModal">
          <SafeAreaView style={styles.popUpContainer}>
            <View style={styles.popUp}>
              <Text style={styles.header}>Are you sure that you want to remove this guardian?</Text>
              <View style={styles.buttonContainer}>
                <TouchableOpacity
                  onPress={() => {
                    removeGuardian(guardianInfo.UserId);
                    setDeleteModalVisible(false);
                  }}
                  style={[styles.button, { backgroundColor: "#F4BFBF" }]}
                >
                  <Text>YES</Text>
                </TouchableOpacity>
                <TouchableOpacity
                  onPress={() => {
                    setDeleteModalVisible(false);
                  }}
                  style={[styles.button, { backgroundColor: "#F4BFBF" }]}
                >
                  <Text>NO</Text>
                </TouchableOpacity>
              </View>
            </View>
          </SafeAreaView>
        </Modal>
      </SafeAreaView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: "row",
    alignItems: "center",
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderBottomWidth: 1,
    borderBottomColor: "#EAEAEA",
  },
  infoContainer: {
    flex: 3,
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "flex-start",
  },
  actionContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    flex: 2,
  },
  name: {
    flex: 3,
    fontSize: 16,
    fontWeight: "bold",
    marginRight: 10,
  },
  email: {
    flex: 4,
    fontSize: 14,
    color: "#888",
  },
  removeButton: {
    backgroundColor: "#BCBCBC",
    borderRadius: 10,
    paddingVertical: 5,
    paddingHorizontal: 10,
    alignItems: "center",
    justifyContent: "center",
  },
  removeText: {
    color: "black",
    fontWeight: "bold",
    fontSize: 14,
  },
  popUpContainer: {
    flex: 1,
    backgroundColor: "rgba(0, 0, 0, 0.8)",
    alignItems: "center",
    justifyContent: "center",
    borderColor: "black",
    borderWidth: 1,
  },
  popUp: {
    backgroundColor: "white",
    width: "90%",
    height: "30%",
    borderRadius: 20,
    alignItems: "center",
    justifyContent: "center",
  },
  buttonContainer: {
    flexDirection: "row",
  },
  header: {
    fontSize: 16,
    marginBottom: 10,
  },
  button: {
    color: "black",
    fontWeight: "bold",
    borderWidth: 1,
    borderColor: "black",
    width: 80,
    alignItems: "center",
    marginHorizontal: 10,
    padding: 10,
    fontSize: 18,
    borderRadius: 40,
  },
});
