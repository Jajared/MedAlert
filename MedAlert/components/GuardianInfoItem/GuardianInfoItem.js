import { MaterialCommunityIcons } from "@expo/vector-icons";
import { StyleSheet, Text, View, SafeAreaView } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";

export default function GuardianInfoItem({ props }) {
  const guardianInfo = props.item;
  return (
    <SafeAreaView style={styles.itemContainer}>
      {guardianInfo.Gender == "Female" ? <MaterialCommunityIcons name="face-woman-profile" size={24} color="black" style={{ flex: 1 }} /> : <MaterialCommunityIcons name="face-man-profile" size={24} color="black" style={{ flex: 1 }} />}
      <View style={styles.infoContainer}>
        <Text style={styles.name}>{guardianInfo.Name}</Text>
        <Text style={styles.email}>{guardianInfo.EmailAddress}</Text>
      </View>
      <View style={styles.actionContainer}>
        <TouchableOpacity style={styles.removeButton} onPress={() => alert("Remove clicked!")}>
          <Text style={styles.removeText}>Remove</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: "row",
    alignItems: "center",
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderBottomWidth: 1,
    borderBottomColor: "#EAEAEA",
  },
  infoContainer: {
    flex: 3,
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "flex-start",
  },
  actionContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    flex: 2,
  },
  name: {
    flex: 3,
    fontSize: 16,
    fontWeight: "bold",
    marginRight: 10,
  },
  email: {
    flex: 4,
    fontSize: 14,
    color: "#888",
  },
  removeButton: {
    backgroundColor: "#BCBCBC",
    borderRadius: 10,
    paddingVertical: 5,
    paddingHorizontal: 10,
    alignItems: "center",
    justifyContent: "center",
  },
  removeText: {
    color: "black",
    fontWeight: "bold",
    fontSize: 14,
  },
});
