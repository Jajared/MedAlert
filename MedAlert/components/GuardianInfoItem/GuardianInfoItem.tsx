import { MaterialCommunityIcons } from "@expo/vector-icons";
import { StyleSheet, Text, View, SafeAreaView, Modal } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import React, { useState } from "react";

export default function GuardianInfoItem({ props, removeGuardian }) {
  const guardianInfo = props.item;
  const [isDeleteModalVisible, setDeleteModalVisible] = useState(false);

  return (
    <SafeAreaView style={styles.itemContainer}>
      {guardianInfo.Gender == "Female" ? <MaterialCommunityIcons name="face-woman-profile" size={24} color="black" style={{ flex: 1 }} /> : <MaterialCommunityIcons name="face-man-profile" size={24} color="black" style={{ flex: 1 }} />}
      <View style={styles.infoContainer}>
        <Text style={styles.name}>{guardianInfo.Name}</Text>
        <Text style={styles.email}>{guardianInfo.EmailAddress}</Text>
      </View>
      <View style={styles.actionContainer}>
        <TouchableOpacity style={styles.removeButton} onPress={() => setDeleteModalVisible(true)}>
          <Text style={styles.removeText}>Remove</Text>
        </TouchableOpacity>
      </View>
      <Modal visible={isDeleteModalVisible} transparent={true} animationType="slide">
        <SafeAreaView style={styles.popUpContainer}>
          <View style={styles.popUp}>
            <Text style={styles.header}>Are you sure that you want to remove this guardian?</Text>
            <View style={styles.buttonContainer}>
              <TouchableOpacity
                onPress={() => {
                  removeGuardian(guardianInfo.UserId);
                  setDeleteModalVisible(false);
                }}
                style={[styles.button, { backgroundColor: "green" }]}
              >
                <Text>YES</Text>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={() => {
                  setDeleteModalVisible(false);
                }}
                style={[styles.button, { backgroundColor: "red" }]}
              >
                <Text>NO</Text>
              </TouchableOpacity>
            </View>
          </View>
        </SafeAreaView>
      </Modal>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: "row",
    alignItems: "center",
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderBottomWidth: 1,
    borderBottomColor: "#EAEAEA",
  },
  infoContainer: {
    flex: 3,
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "flex-start",
  },
  actionContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    flex: 2,
  },
  name: {
    flex: 3,
    fontSize: 16,
    fontWeight: "bold",
    marginRight: 10,
  },
  email: {
    flex: 4,
    fontSize: 14,
    color: "#888",
  },
  removeButton: {
    backgroundColor: "#BCBCBC",
    borderRadius: 10,
    paddingVertical: 5,
    paddingHorizontal: 10,
    alignItems: "center",
    justifyContent: "center",
  },
  removeText: {
    color: "black",
    fontWeight: "bold",
    fontSize: 14,
  },
  popUpContainer: {
    flex: 1,
    backgroundColor: "rgba(0, 0, 0, 0.8)",
    alignItems: "center",
    justifyContent: "center",
    borderColor: "black",
    borderWidth: 1,
  },
  popUp: {
    backgroundColor: "white",
    width: "90%",
    height: "30%",
    borderRadius: 20,
    alignItems: "center",
    justifyContent: "center",
  },
  buttonContainer: {
    flexDirection: "row",
  },
  header: {
    fontSize: 16,
    marginBottom: 10,
  },
  button: {
    color: "black",
    fontWeight: "bold",
    borderWidth: 1,
    borderColor: "black",
    width: 80,
    alignItems: "center",
    marginHorizontal: 10,
    padding: 10,
    fontSize: 18,
    borderRadius: 40,
  },
});
