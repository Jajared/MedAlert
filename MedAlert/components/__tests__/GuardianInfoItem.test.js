import { render, screen, fireEvent } from "@testing-library/react-native";
import GuardianInfoItem from "../GuardianInfoItem";
import { expect, jest, test } from "@jest/globals";

jest.mock("expo-font");
jest.mock("expo-asset");

const navigationObject = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

const testGuardianInfo = {
  UserId: "123",
  Name: "test",
  Gender: "male",
  DateOfBirth: "01/01/2000",
  EmailAddress: "test@gmail.com",
  PhoneNumber: "12345678",
  DeviceToken: "12345678",
};

describe("GuardianInfoItem", () => {
  const removeGuardian = jest.fn();
  // Check if renders correctly
  it("renders correctly", () => {
    const tree = render(<GuardianInfoItem navigation={navigationObject} props={{ item: testGuardianInfo }} removeGuardian={removeGuardian} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Check number of children
  it("has 2 children", () => {
    render(<GuardianInfoItem navigation={navigationObject} props={{ item: testGuardianInfo }} removeGuardian={removeGuardian} />);
    expect(screen.getByTestId("GuardianInfoItem").children.length).toBe(2);
  });
  // Check if navigates to patient medication page when pressed
  it("Navigates to patient medication page", () => {
    render(<GuardianInfoItem navigation={navigationObject} props={{ item: testGuardianInfo }} removeGuardian={removeGuardian} />);
    fireEvent.press(screen.getByTestId("GuardianInfoItem"));
    expect(navigationObject.navigate).toBeCalledWith("Patient Medication", { guardianId: testGuardianInfo.UserId, guardianName: testGuardianInfo.Name });
  });
  // Check if delete modal pops up when remove button is pressed
  it("Delete modal pops up", () => {
    render(<GuardianInfoItem navigation={navigationObject} props={{ item: testGuardianInfo }} removeGuardian={removeGuardian} />);
    fireEvent.press(screen.getByTestId("RemoveButton"));
    expect(screen.getByTestId("DeleteModal")).toBeTruthy();
  });
});
