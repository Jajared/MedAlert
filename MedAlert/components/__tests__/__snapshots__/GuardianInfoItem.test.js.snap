// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuardianInfoItem renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="GuardianInfoItem"
>
  <RCTSafeAreaView
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#EAEAEA",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
  >
    <Text />
    <View
      style={
        {
          "alignItems": "flex-start",
          "flex": 3,
          "flexDirection": "column",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          {
            "flex": 3,
            "fontSize": 16,
            "fontWeight": "bold",
            "marginRight": 10,
          }
        }
      >
        test
      </Text>
      <Text
        style={
          {
            "color": "#888",
            "flex": 4,
            "fontSize": 14,
          }
        }
      >
        test@gmail.com
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 2,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#BCBCBC",
            "borderRadius": 10,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
        testID="RemoveButton"
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Remove
        </Text>
      </View>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      testID="DeleteModal"
      transparent={true}
      visible={false}
    />
  </RCTSafeAreaView>
</View>
`;
